src/test/java/uk/gov/hmcts/reform/civil/service/hearings/HearingFeesServiceTest.java

```diff
@@ -0,0 +1,155 @@
+        // Given
+        given(restTemplate.getForObject(queryCaptor.capture(), eq(FeeLookupResponseDto.class)))
+            .willReturn(FeeLookupResponseDto.builder()
+                            .feeAmount(TEST_FEE_AMOUNT_POUNDS_14)
+                            .code("test_fee_code")
+                            .version(2)
+                            .build());
+
+        BigDecimal claimAmount = new BigDecimal(125);
+
+        Fee expectedFeeDto = Fee.builder()
+            .calculatedAmountInPence(TEST_FEE_AMOUNT_PENCE_14)
+            .code("test_fee_code")
+            .version("2")
+            .build();
+        // When
+        Fee feeDto = feesService.getFeeForHearingFastTrackClaims(claimAmount);
+        // Then
+        assertThat(feeDto).isEqualTo(expectedFeeDto);
+        assertThat(queryCaptor.getValue().toString())
+            .isEqualTo("dummy_url/fees-register/fees/lookup?channel=default&event=hearing&jurisdiction1"
+                           + "=civil&jurisdiction2=civil&service=civil%20money%20claims"
+                           + "&keyword=FastTrackHrgKey&amount_or_volume=125");
+    }
+
+    @Test
+    void throwRuntimeException_whenFeeServiceThrowsException() {
+        // Given
+        given(restTemplate.getForObject(queryCaptor.capture(), eq(FeeLookupResponseDto.class)))
+            .willThrow(new RestClientException("No Fees returned by fee-service while creating hearing fee"));
+
+        BigDecimal claimAmount = new BigDecimal(125);
+        // When
+        Exception exception = assertThrows(
+            RestClientException.class, () -> feesService.getFeeForHearingSmallClaims(claimAmount));
+        // Then
+        assertThat(exception).isInstanceOf(RestClientException.class);
+    }
+
+    @Test
+    void throwRuntimeException_whenNoFeeIsReturnedByFeeService() {
+        // Given
+        given(restTemplate.getForObject(queryCaptor.capture(), eq(FeeLookupResponseDto.class)))
+            .willReturn(null);
+
+        BigDecimal claimAmount = new BigDecimal(125);
+
+        Exception exception = assertThrows(
+            RestClientException.class, () -> feesService.getFeeForHearingSmallClaims(claimAmount));
+        // Then
+        assertThat(exception.getMessage())
+            .isEqualTo("No Fees returned by fee-service while creating hearing fee");
+    }
+
+    @Test
+    void throwRuntimeException_whenNoFeeAmountIsReturnedByFeeService() {
+        // Given
+        given(restTemplate.getForObject(queryCaptor.capture(), eq(FeeLookupResponseDto.class)))
+            .willReturn(FeeLookupResponseDto.builder()
+                            .code("test_fee_code")
+                            .version(1)
+                            .build());
+
+        BigDecimal claimAmount = new BigDecimal(125);
+        // When
+        Exception exception = assertThrows(
+            RestClientException.class, () -> feesService.getFeeForHearingSmallClaims(claimAmount));
```

Comment:
The following is needed if you apply other suggestions.

```suggestion
            InternalServerErrorException.class, () -> feesService.getFeeForHearingSmallClaims(claimAmount));
```